### PR TITLE
Test refactor

### PR DESCRIPTION
**Pull request by bamandel**
Added tests for the Processor package as well as some fixes to my previous test refactoring.

### CODE REVIEW

File: src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/distribution/EmailConfigEntityTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import com.blackducksoftware.integration.hub.alert.mock.EmailMockUtils;
+import com.blackducksoftware.integration.hub.alert.mock.MockUtils;
+
+public class EmailConfigEntityTest extends EntityTest<EmailGroupDistributionConfigEntity> {
+    private final EmailMockUtils mockUtils = new EmailMockUtils();
+
+    @Override
+    public MockUtils<?, ?, EmailGroupDistributionConfigEntity, ?> getMockUtil() {
+        return mockUtils;
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/distribution/EmailConfigEntityTest.java]: You could just do 'new EmailMockUtils()' in the getMockUtil

File: src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/distribution/HipChatConfigEntityTest.java [hunk 2]
```diff
@@ -15,13 +15,19 @@
...
 public class HipChatConfigEntityTest extends EntityTest<HipChatDistributionConfigEntity> {
-    private static final HipChatMockUtils mockUtils = new HipChatMockUtils();
+    private final HipChatMockUtils mockUtils = new HipChatMockUtils();
 
-    // TODO create an abstract method to set MockUtils to avoid static (Do this everywhere necessary)
-    public HipChatConfigEntityTest() {
-        super(mockUtils, HipChatDistributionConfigEntity.class);
+    @Override
+    public MockUtils<?, ?, HipChatDistributionConfigEntity, ?> getMockUtil() {
+        return mockUtils;
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/distribution/HipChatConfigEntityTest.java]: You could just do 'new HipChatMockUtils()' in the getMockUtil

File: src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/distribution/SlackConfigEntityTest.java [hunk 3]
```diff
@@ -14,14 +14,15 @@
...
 
 public class SlackConfigEntityTest extends EntityTest<SlackDistributionConfigEntity> {
-    private static final SlackMockUtils mockUtils = new SlackMockUtils();
+    private final SlackMockUtils mockUtils = new SlackMockUtils();
 
-    public SlackConfigEntityTest() {
-        super(mockUtils, SlackDistributionConfigEntity.class);
+    @Override
+    public MockUtils<?, ?, SlackDistributionConfigEntity, ?> getMockUtil() {
+        return mockUtils;
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/distribution/SlackConfigEntityTest.java]: You could just do 'new SlackMockUtils()' in the getMockUtil

File: src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/global/GlobalHubConfigEntityTest.java [hunk 4]
```diff
@@ -14,53 +14,48 @@
...
-
-        assertEquals(31860737, globalConfigEntity.hashCode());
-        final String expectedString = "{\"hubTimeout\":null,\"hubUsername\":null,\"id\":null}";
-        assertEquals(expectedString, globalConfigEntity.toString());
-
-        final GlobalHubConfigEntity globalConfigEntityNew = new GlobalHubConfigEntity();
-        assertEquals(globalConfigEntity, globalConfigEntityNew);
+    @Override
+    public MockUtils<?, ?, ?, GlobalHubConfigEntity> getMockUtil() {
+        return mockUtils;
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/global/GlobalHubConfigEntityTest.java]: You could just do 'new GlobalHubMockUtils()' in the getMockUtil

File: src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/global/GlobalSchedulingConfigEntityTest.java [hunk 5]
```diff
@@ -52,4 +49,14 @@ public int globalEntityHashCode() {
         return -1636279562;
     }
 
+    @Override
+    public MockUtils<?, ?, ?, GlobalSchedulingConfigEntity> getMockUtil() {
+        return mockUtils;
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/global/GlobalSchedulingConfigEntityTest.java]: You could just do 'new GlobalSchedulingMockUtils()' in the getMockUtil

File: src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyOverrideProcessorTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        final ProjectVersionModel projectVersionModel = new ProjectVersionModel();
+        final String componentName = "Content item test";
+        final ComponentVersionView componentVersionView = new ComponentVersionView();
+        final String componentUrl = "google.com";
+        final String componentVersionUrl = "newest";
+        final PolicyRuleView policyRuleView = new PolicyRuleView();
+        final List<PolicyRuleView> policyRuleList = Arrays.asList(policyRuleView);
+        final String firstName = "B";
+        final String lastName = "Dawg";
+        final String coponentIssueUrl = "broken.edu";
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyOverrideProcessorTest.java]: componentIssueUrl - missed the m

File: src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyViolationClearedProcessorTest.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+        final Date createdAt = new Date();
+        final ProjectVersionModel projectVersionModel = new ProjectVersionModel();
+        final String componentName = "Content item test";
+        final ComponentVersionView componentVersionView = new ComponentVersionView();
+        final String componentUrl = "url";
+        final String componentVersionUrl = "newest";
+        final PolicyRuleView policyRuleView = new PolicyRuleView();
+        final List<PolicyRuleView> policyRuleList = Arrays.asList(policyRuleView);
+        final String coponentIssueUrl = "broken.edu";
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyViolationClearedProcessorTest.java]: componentIssueUrl - missed the m

File: src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyViolationProcessorTest.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        final String componentName = "Content item test";
+        final ComponentVersionView componentVersionView = new ComponentVersionView();
+        final String componentUrl = "url";
+        final String componentVersionUrl = "newest";
+        final PolicyRuleView policyRuleView = new PolicyRuleView();
+        final MetaView metaView = new MetaView();
+        metaView.href = "Meta href";
+        policyRuleView.meta = metaView;
+        final List<PolicyRuleView> policyRuleList = Arrays.asList(policyRuleView);
+        final String coponentIssueUrl = "issueUrl";
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/processor/PolicyViolationProcessorTest.java]: componentIssueUrl - missed the m

File: src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/distribution/SlackConfigEntityTest.java (hunk 3 above)
Comment (jamesrichard91) [reply src/test/java/com/blackducksoftware/integration/hub/alert/datasource/entity/distribution/SlackConfigEntityTest.java]: Will java allow you to change the return type to SlackMockUtils and still be a valid override for the abstract method?

File: src/test/java/com/blackducksoftware/integration/hub/alert/web/model/global/GlobalHipChatConfigRestModelTest.java [hunk 9]
```diff
@@ -47,4 +44,14 @@ public void assertGlobalRestModelFieldsFull(final GlobalHipChatConfigRestModel r
     public int globalRestModelHashCode() {
         return 608471908;
     }
+
+    @Override
+    public MockUtils<?, GlobalHipChatConfigRestModel, ?, ?> getMockUtil() {
+        return mockUtils;
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/web/model/global/GlobalHipChatConfigRestModelTest.java]: new HipChatMockUtils()

File: src/test/java/com/blackducksoftware/integration/hub/alert/web/model/global/GlobalHubConfigRestModelTest.java [hunk 10]
```diff
@@ -58,11 +54,20 @@ public void assertGlobalRestModelFieldsFull(final GlobalHubConfigRestModel restM
...
     }
 
     @Override
     public int globalRestModelHashCode() {
         return 1536081842;
     }
+
+    @Override
+    public MockUtils<?, GlobalHubConfigRestModel, ?, ?> getMockUtil() {
+        return mockUtils;
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/web/model/global/GlobalHubConfigRestModelTest.java]: new GlobalHubMockUtils()

File: src/test/java/com/blackducksoftware/integration/hub/alert/web/model/global/GlobalSchedulingConfigRestModelTest.java [hunk 11]
```diff
@@ -52,4 +49,14 @@ public int globalRestModelHashCode() {
         return -1636279514;
     }
 
+    @Override
+    public MockUtils<?, GlobalSchedulingConfigRestModel, ?, ?> getMockUtil() {
+        return mockUtils;
```
Comment (jamesrichard91) [inline src/test/java/com/blackducksoftware/integration/hub/alert/web/model/global/GlobalSchedulingConfigRestModelTest.java]: new GlobalSchedulingMockUtils()

**Comment by bamandel** [comment]
Have a new branch with further changes as well as the ones recommended by James

**Pull request closed without merging**